calculator/math_parser.py

```diff
@@ -104,7 +108,8 @@ def evaluate(expression):
     """
     op_stack  = deque() # operator stack
     out_stack = deque() # output stack (values)
-    for token in parse(expression):
+    tokens = parse(expression) # calls the function only once!
+    for token in tokens:

```

Comment:
Actually, the expression to the right of `for` loop is evaluated only once. Hence no need for this.

Reply:
Ok, I understand.